exe/Main.hs

```diff
@@ -78,8 +78,11 @@ main = do
     command <- makeLspCommandId "typesignature.add"
 
     let plugins = Completions.plugin <> CodeAction.plugin
-        onInitialConfiguration = const $ Right ()
-        onConfigurationChange  = const $ Right ()
+        
+        -- we use `Maybe InitializeParams` as the config type.
+        -- Change this if you want to parse InitializeRequest/DidChangeConfigurationNotification differently.
+    let onInitialConfiguration RequestMessage{_params=initParams} = Right $ Just initParams
+        onConfigurationChange = const $ Right Nothing
```

Comment:
The `config` type is the settings that users can customise from the language clients, not something internal to the language server. `haskell-lsp` handles this a bit weirdly in that it lets us parse it ourselves from the initialisation message here but it's probably best to not use it as a mechanism for passing about information from the initialization request